docs/user_guide/building_blocks/categories.md

```diff
@@ -1,3 +1,495 @@
+    --calibrators example --selector example --producer category_ids \
+    --processes tt,st --categories "*"
+```
+
+
+As a next step, one might want to combine these categories to categorize based on the number of
+jets and leptons simutaneously. You could simply create all combinations by hand. The following code
+snippet shows how to combine the "2jet" and the "1lep" category:
+
+```python
+cat_2jet = config.get_category("2jet")
+cat_1lep = config.get_category("1lep")
+
+# add combined category as child to the existing categories
+cat_1e__2jet = cat_2jet.add_category(
+    name="1e__2jet",
+    id=cat_2jet.id + cat_1e.id,
+    # our `category_ids` producer can automatically combine multiple selections via logical 'and'
+    selection=[cat_1e.selection, cat_2jet.selection],
+    label="1 electron, 0 muons, >2 jets",
+)
+cat_1mu__2jet = cat_2jet.add_category(
+    name="1e__2jet",
+    id=cat_2jet.id + cat_1mu.id,
+    selection=[cat_1mu.selection, cat_2jet.selection],
+    label="1 muon, 0 electrons, 2 jets",
+)
+cat_1e__3jet = cat_3jet.add_category(
+    name="1e__3jet",
+    id=cat_3jet.id + cat_1e.id,
+    selection=[cat_1e.selection, cat_3jet.selection],
+    label="1 electron, 0 muons, 3 jets",
+)
+cat_1mu__3jet = cat_3jet.add_category(
+    name="1e__3jet",
+    id=cat_3jet.id + cat_1mu.id,
+    selection=[cat_1mu.selection, cat_3jet.selection],
+    label="1 muon, 0 electrons, 3 jets",
+)
+
+# add childs also to lepton categories
+cat_1e.add_category(cat_1e__2jet)
+cat_1e.add_category(cat_1e__3jet)
+cat_1mu.add_category(cat_1mu__2jet)
+cat_1mu.add_category(cat_1mu__3jet)
+```
+
+We now created categories with dependencies between each other. In columnflow, we always use the
+smallest units of categories (commonly called "leaf categories") to build our parent categories. For
+example, the `2jet` category consists of the leaf categories `1e__2jet` and `1mu__2jet`. To select
+events corresponding to `2jet`, we will add all events with category ids corresponding to either
+`1e__2jet` or `1mu__2jet`.
+
+
+:::{note} We might add unexpected selections by combining categorization.
+
+In this example, the `cat_1e` and `cat_1mu` Categorizer only select events with exactly one lepton
+(either electron or muon). This means, that events with zero or multiple leptons will not be considered
+in our categorization, event when building e.g. the `2jet` category.
+:::
+
+Let's test if our leaf categories are working as intended:
+
+```shell
+law run cf.CreateYieldsTable --version v1 \
+    --calibrators example --selector example --producer category_ids \
+    --processes tt,st --categories 1e,2jet,1e__2jet,1e__3jet,1mu__2jet,1mu__3jet
+```
+
+We should see that the yield of the `2jet` category is the same as the `1e__2jet` and `1mu__2jet`
+categories combined. This is to be expected, since the `2jet` category will be built by combining
+histograms from all of their leaf categories.
+
+:::{dropdown} Note! Take care to define your categories orthogonal!
```

Comment:
I don't really want to extend all of this by default because this contains a code example which nobody should use in their analysis. I changed the first part to a note and added the code example as a dropdown to the note